src/components/error-message.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { Text, Card, CardItem, Body } from 'native-base';
+import { Animated, Easing } from "react-native";
+
+import Styles from '../styles/main';
+
+interface Props {
+  error: string;
+}
+
+const errInicialPosition = -58;
+const errEndPosition = 0;
+const errPositionDuration = 150;
+
+const errInitialOpacity = 0;
+const errEndOpacity = 1;
+const errOpacityDuration = 0;
+
+interface State {
+  errOpacity: Animated.Value;
+  errPosition: Animated.Value;
+}
+
+export class ErrorMessage extends React.Component <Props, State> {
+  state = {
+    errOpacity: new Animated.Value(errInitialOpacity),
+    errPosition: new Animated.Value(errInicialPosition)
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    const { error } = this.props
+    const { error: nextError } = nextProps;
+
+    return !(!error && !nextError) && (error !== nextError);
```

Comment:
!(!error && !nextError) is the same as error || nextError, which is much cleaner:
```javascript
return (error || nextError) && error !== nextError
```